### PR TITLE
Rule engine refactoring

### PR DESCRIPTION
**Pull request by JindrichSusen**

### CODE REVIEW

File: backend/Origam.Schema.EntityModel/XsltFunctionSchemaItemProvider.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        public override Type[] NewItemTypes
+        {
+            get
+            {
+                return new Type[1] {typeof(XsltFunctionCollection)};
+            }
+        }
+
+        public override AbstractSchemaItem NewItem(Type type, Guid schemaExtensionId, SchemaItemGroup group)
```
Comment (tvavrda) [inline backend/Origam.Schema.EntityModel/XsltFunctionSchemaItemProvider.cs:54]: You may ommit `NewItem` and `NewItemTypes` methods by adding `this.ChildItemTypes.Add(typeof(XsltFunctionCollection))` to the constructor.

See `AbstractWorkflowBlock` for an example.

**Review by tvavrda — CHANGES_REQUESTED** [review]

File: backend/Origam.Workbench.Services/CoreServices/DataServiceFactory.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
+﻿#region license
```
Comment (washibana) [inline backend/Origam.Workbench.Services/CoreServices/DataServiceFactory.cs:1]: This is a new file, so license years should reflect this year (2022).

File: backend/Origam.Workbench.Services/CoreServices/ICoreDataService.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+﻿using System;
```
Comment (washibana) [inline backend/Origam.Workbench.Services/CoreServices/ICoreDataService.cs:1]: Missing license information.

File: backend/Origam.Workflow/Origam.Workflow.csproj [hunk 4]
```diff
@@ -94,6 +94,7 @@
     <Compile Update="WorkQueue\WorkQueueData.Designer.cs">
       <SubType>Code</SubType>
     </Compile>
+    <Compile Remove="NullStateMachineService.cs" />
```
Comment (washibana) [inline backend/Origam.Workflow/Origam.Workflow.csproj:97]: Please review the necessity of the instruction.

**Review by washibana — CHANGES_REQUESTED** [review]

**Comment by washibana** [comment]
@JindrichSusen We also need an article on support for external collections of XSLT functions.

**Review by washibana — APPROVED** [review]

**Pull request merged**